src/dsa/rpo_stark/stark/mod.rs

```diff
@@ -67,3 +75,36 @@ where
         )
     }
 }
+
+/// Deterministically generates a seed for seeding the PRNG used for zero-knowledge.
+///
+/// This uses the Algorithm described in [RFC 6979](https://tools.ietf.org/html/rfc6979#section-3) § 3.2.
+/// The direct approach would be to just use the concatentation of the secret key and the message as
+/// the value of the seed but we opt instead to use it as the seed of an `HMAC_DRBG` PRNG similar
+/// to how it is used in `RFC 6979` to generate the value `k`.
+///
+/// Note that in `RFC 6979` the hash function used in the `HMAC_DRBG` PRNG is chosen to be the same
+/// hash function used in hashing the message. In Section 3.6., however, a variant allowing
+/// different hash functions is discussed and the overall security is claimed to be limited by
+/// the weaker of the two.
+#[inline]
+pub fn generate_seed<D, N>(
```

Comment:
Very cool! If I understood this correctly, we are basically computing the seed as `hash(sk || msg)`, right? Or is there  more to it?

Reply:
That's indeed correct but just wanted to be conservative and use the construction in the reference.
Should we just simplify it?

Reply:
If it is this simple, I think it probably would be better to avoid the extra dependency and just implement the algorithm ourselves (the reference to the algorithm would probably be sufficient).

Reply:
Can we somehow argue that `hash(sk || msg)` is equivalent to this construction (or at least has the same properties)? I'm curious why RFC6979 needs all this extra complexity vs. just computing a simple hash.

Reply:
The main difference I see is that in RFC6979 the end goal is to generate a random element, i.e., `k`, in the field.
To this end, one sort of uses `hash(sk || msg)` as the seed of a PRNG. This PRNG is HMAC based and uses the same hash function that was used in `hash(sk || msg)`.
In other words, RFC6979 gives a method to build a PRNG, seed it and then generate an element in a given field, all of this while only relying on the same hash function used to process the message i.e., reduce the number of assumptions.

In our use case things are different, we only need to seed **a** PRNG, and hence we can use the same idea above, namely `hash(sk || msg)` which amounts to using `sk` for entropy and `msg` for a nonce effect.

One thing which we will have to do here, and in fact in other places as well, is to domain separate in `hash(sk || msg)`, maybe something like `hash(sk || msg || STRING)`  where `STRING` is a string that helps to uniquely identify our signature scheme e.g., name + security level. In fact, the BCS transform assumes that for each round of the IOP the random oracle used is different. Hence, for each round of an IOP we will have to domain separate when absorbing.